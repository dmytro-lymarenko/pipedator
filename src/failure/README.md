- **Description**

Creates a validator which always fails (the opposite to [`success`](#success)).

- **How to import**

```typescript
import { failure } from 'pipedator';
// or
import { failure } from 'pipedator/lib/failure';

```
- **Signature**

```typescript
function failure<ValidValue = any>(message?: string): Validator<ValidValue>;
```
- **Parameters**

  - `message` - (optional) custom message.


- **Usage**

```typescript
failure().validate(1 /* or any other value */); // invalid
failure('Custom message').validate(1 /* or any other value */); // invalid with 'Custom message'
```