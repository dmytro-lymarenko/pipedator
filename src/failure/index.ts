/**
 * @module validators
 */

import { createValidator } from '../core';

export function failure(message?: string) {
	return createValidator({
		validate: (value, ctx) =>
			ctx.generateError({
				value,
				message: message || 'Should always fail',
				path: ctx.path,
			}),
	});
}
