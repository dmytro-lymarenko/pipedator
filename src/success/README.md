- **Description**

Creates a validator which always fails (the opposite to [`failure`](#failure)).

- **How to import**

```typescript
import { success } from 'pipedator';
// or
import { success } from 'pipedator/lib/success';

```
- **Signature**

```typescript
function success<ValidValue = any>(): Validator<ValidValue>;
```

- **Usage**

```typescript
success().validate(1 /* or any other value */); // valid
```