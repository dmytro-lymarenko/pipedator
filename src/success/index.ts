import { createValidator } from '../core';

export function success() {
	return createValidator({
		validate: () => null,
	});
}
