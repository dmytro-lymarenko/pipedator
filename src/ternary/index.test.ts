import { ternary } from './index';
import { success } from '../success';
import { failure } from '../failure';

describe('ternary()', () => {
	it('should validate when both condition and success succeed', () => {
		expect(ternary(success(), success(), success()).validate(1)).toBeNull();
		expect(ternary(success(), success(), failure()).validate(1)).toBeNull();
	});

	it('should validate when condition fails but failure succeed', () => {
		expect(ternary(failure(), success(), success()).validate(1)).toBeNull();
		expect(ternary(failure(), failure(), success()).validate(1)).toBeNull();
	});

	it('should fail with correct message when condition succeeds but success fails', () => {
		expect(ternary(success(), failure('Correct message'), success()).validate(1)).toMatchSnapshot();
		expect(ternary(success(), failure('Correct message'), failure('Incorrect message')).validate(1)).toMatchSnapshot();
	});

	it('should fail with correct message when both condition and failure fail', () => {
		expect(ternary(failure('Incorrect message'), success(), failure('Correct message')).validate(1)).toMatchSnapshot();
		expect(
			ternary(failure('Incorrect message'), failure('Incorrect message'), failure('Correct message')).validate(1)
		).toMatchSnapshot();
	});
});
