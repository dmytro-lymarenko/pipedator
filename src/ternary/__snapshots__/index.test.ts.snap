// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ternary() should fail with correct message when both condition and failure fail 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Correct message",
      "path": Array [],
      "value": 1,
    },
  ],
  "rootValue": 1,
}
`;

exports[`ternary() should fail with correct message when both condition and failure fail 2`] = `
Object {
  "errors": Array [
    Object {
      "message": "Correct message",
      "path": Array [],
      "value": 1,
    },
  ],
  "rootValue": 1,
}
`;

exports[`ternary() should fail with correct message when condition succeeds but success fails 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Correct message",
      "path": Array [],
      "value": 1,
    },
  ],
  "rootValue": 1,
}
`;

exports[`ternary() should fail with correct message when condition succeeds but success fails 2`] = `
Object {
  "errors": Array [
    Object {
      "message": "Correct message",
      "path": Array [],
      "value": 1,
    },
  ],
  "rootValue": 1,
}
`;
